server/src/main/java/org/spine3/server/entity/VisibleEntity.java

```diff
@@ -18,27 +18,24 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-package org.spine3.change;
+package org.spine3.server.entity;
 
-import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Preconditions.checkNotNull;
+import com.google.protobuf.Message;
 
 /**
- * Routines common for all mismatches.
+ * An entity which has visibility.
  *
  * @author Alexander Yevsyukov
  */
-class Mismatches {
+public interface VisibleEntity<I, S extends Message> extends Entity<I, S> {
 
-    private static final String ERR_CANNOT_BE_EQUAL = "`expected` and `actual` cannot be equal in ValueMismatch";
+    /**
+     * Obtains the current visibility status.
+     */
+    Visibility getVisibility();
 
-    private Mismatches() {
-        // Prevent instantiations.
-    }
-
-    static void checkNotNullOrEqual(Object expected, Object actual) {
-        checkNotNull(expected);
-        checkNotNull(actual);
-        checkArgument(!expected.equals(actual), ERR_CANNOT_BE_EQUAL);
-    }
+    /**
+     * Verifies whether visibility of the entity changed since its initialization.
+     */
+    boolean visibilityChanged();
```

Comment:
That's an odd interface method. Seems more like a utility method of an `abstract class`-level.